### PR TITLE
Update auto-generated spec file so it actually works

### PR DESCRIPTION
**Pull request by gicmo**
The gcc/gcc-c++ compilers are not in the buildroot anymore therefore
we need to explicitly require them while building.
Additionally fix the download URL to something that would work.

If one uses Makefiles, then in the `<builddir>` a rpm can be build with (given build-dependencies are installed and the current release is F29):

```
make package_source
fedpkg --release F29 local
```

### CODE REVIEW

**Comment by jgrewe** [comment]
so we can kill the nixio.spec file from the repo?

**Comment by gicmo** [comment]
@jgrewe could do. We can keep the `nixio.spec` in 1.4 branch from where we can build stable release on copr for 1.4. Once we have 1.5.0 out we should try to get it into fedora so we don't need the non-generated one anymore for 1.5. What do you think?

File: nixio.spec.in [hunk 1]
```diff
@@ -5,9 +5,13 @@ Summary:	IO-libray for nix data files
 
 License:	BSD
 URL:		https://www.g-node.org/nix
-Source0:	%{name}-%{version}.tar.gz
+Source0:	https://github.com/G-Node/nix/archive/%{version}/%{name}-%{version}.tar.gz
```
Comment (jgrewe) [inline nixio.spec.in]: if I am not mistaken then the tar url is https://github.com/G-Node/nix/archive/%{version}.tar.gz

**Comment by jgrewe** [comment]
@gicmo, Yes, that is what I had in mind. Removing it only from the master branch. In the 1.4 we might still use it for further reference

File: nixio.spec.in (hunk 1 above)
Comment (jgrewe) [reply nixio.spec.in]: hm, not sure, If I try curl on the urls, it does not work. ``wget https://github.com/G-Node/nix/archive/1.4.3/1.4.3.tar.gz`` works but is redirected to `` https://codeload.github.com/G-Node/nix/tar.gz/1.4.3`` so ``curl https://codeload.github.com/G-Node/nix/tar.gz/1.4.3 -o 1.4.3.tar.gz`` succeeds.
So, imho, the ``%{name}-%{version}.tar.gz`` part will not work.

Comment (gicmo) [reply nixio.spec.in]: Edit the generated spec-file to make the url point to something valid:
```
--- nixio.spec	2019-01-06 13:04:15.465030789 +0100
+++ nixio-1.4.3.spec	2019-01-06 13:04:01.057849027 +0100
@@ -1,5 +1,5 @@
 Name:		nixio
-Version:	1.5.0
+Version:	1.4.3
 Release:	0.20190104gitcc8d3764%{?dist}
 Summary:	IO-libray for nix data files
```
Then use `spectool` to try to download it directly from the specfile:
```
spectool -g -R -C . nixio.spec
Getting https://github.com/G-Node/nix/archive/1.4.3/nixio-1.4.3.tar.gz to ./nixio-1.4.3.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   117    0   117    0     0    194      0 --:--:-- --:--:-- --:--:--   194
100  463k    0  463k    0     0   271k      0 --:--:--  0:00:01 --:--:-- 1672k
```

**Pull request merged**